lib/ruby_lsp/response_builders/hover.rb

```diff
@@ -12,27 +12,37 @@ class Hover < ResponseBuilder
       sig { void }
       def initialize
         super
-        @stack = T.let(
-          [],
-          T::Array[String],
+
+        @response = T.let(
+          {
+            title: +"",
+            links: +"",
+            documentation: +"",
+          },
+          T::Hash[Symbol, String],
         )
       end
 
-      sig { params(hover_response: String).void }
-      def push(hover_response)
-        @stack << hover_response
+      sig { params(content: String, category: Symbol).void }
+      def push(content, category)
+        hover_content = @response[category]
+        if hover_content
+          hover_content << content + "\n"
+        end
```

Comment:
Should we raise if developers try to push to an not supported category?

Reply:
I would vote for just ignoring it so that addons don't end up breaking hover.

Reply:
Yeah, we ultimately don't want the core Ruby LSP hover to be vulnerable to disruption by addons.